src/ops/verify_proof.rs

```diff
@@ -0,0 +1,140 @@
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use super::AbiCall;
+use crate::call_context::{CallContext, Resolver};
+use crate::VMError;
+
+use canonical::Store;
+use dusk_plonk::prelude::*;
+use transfer_circuits::{
+    ExecuteCircuit, SendToContractObfuscatedCircuit,
+    SendToContractTransparentCircuit, WithdrawFromObfuscatedCircuit,
+};
+use wasmi::{RuntimeArgs, RuntimeValue, ValueType};
+
+pub struct ProofVerification;
+
+impl<E: Resolver<S>, S: Store> AbiCall<E, S> for ProofVerification {
+    const ARGUMENTS: &'static [ValueType] = &[
+        ValueType::I32,
+        ValueType::I32,
+        ValueType::I32,
+        ValueType::I32,
+        ValueType::I32,
+        ValueType::I32,
+    ];
+    const RETURN: Option<ValueType> = Some(ValueType::I32);
+
+    fn call(
+        context: &mut CallContext<E, S>,
+        args: RuntimeArgs,
+    ) -> Result<Option<RuntimeValue>, VMError<S>> {
+        if let [RuntimeValue::I32(pub_inp), RuntimeValue::I32(pub_inp_len), RuntimeValue::I32(proof), RuntimeValue::I32(verif_key), RuntimeValue::I32(label), RuntimeValue::I32(label_len)] =
+            args.as_ref()[..]
+        {
+            let pub_inp = pub_inp as usize;
+            let pub_inp_len = pub_inp_len as usize;
+            let proof = proof as usize;
+            let verif_key = verif_key as usize;
+            let label = label as usize;
+            let label_len = label_len as usize;
+
+            context.memory(|mem| -> Result<Option<RuntimeValue>, VMError<S>> {
+                let pi_vec = decode_pub_inputs(mem, pub_inp_len, pub_inp)?;
+                println!("PIVEC");
+                let proof = Proof::from_bytes(
+                    &mem[proof..proof + Proof::serialised_size()],
+                )
+                .map_err(|_| VMError::InvalidArguments)?;
+                println!("proof");
+                let vk = VerifierKey::from_bytes(
+                    &mem[verif_key..verif_key + VerifierKey::serialised_size()],
+                )
+                .map_err(|_| VMError::InvalidArguments)?;
+                println!("vk");
+                let label =
+                    String::from_utf8(mem[label..label + label_len].to_vec())
+                        .map_err(|_| VMError::InvalidUtf8)?;
+                println!("label");
+                let pp = unsafe {
+                    PublicParameters::from_slice_unchecked(
+                        &rusk_profile::get_common_reference_string().map_err(
+                            |_| {
+                                VMError::ContractPanic(
+                                    "PubParams deser error".to_string(),
+                                )
+                            },
+                        )?,
+                    )
+                    .map_err(|_| VMError::InvalidArguments)?
+                };
+                println!("PP");
+                match select_and_verify::<S>(&label, &pp, &vk, &pi_vec, &proof)
```

Comment:
I don't see why `select_and_verify` cannot returns directly a boolean as well. In that case you can return directly that avoiding the match:
```rust
let succeed = select_and_verify::<S>(…);
Ok(Some(RuntimeValue::I32(succeed as i32)))
```